Final Legal Chatbot Files/vector_store.py

```
@@ -0,0 +1,171 @@
+    # Stage 5 — generate answer
+    try:
+        if DEFAULT_LLM == "openai":
+            response = llm_engine.generate_response(
+                question=standardized_query, context=context, role=role,
+            )
+        else:
+            response = llm_engine.generate_response(
+                question=standardized_query, context=context,
+            )
+        return response.strip(), citations
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)